chainservice/chainservice.go

```diff
@@ -165,12 +179,8 @@ func New(
 		cfg.DB.DbPath = cfg.Chain.ChainDBPath
 		kvStore = db.NewBoltDB(cfg.DB)
 	}
-	var dao blockdao.BlockDAO
-	if gateway && !cfg.Chain.EnableAsyncIndexWrite {
-		dao = blockdao.NewBlockDAO(kvStore, []blockdao.BlockIndexer{indexer}, cfg.Chain.CompressBlock, cfg.DB)
-	} else {
-		dao = blockdao.NewBlockDAO(kvStore, nil, cfg.Chain.CompressBlock, cfg.DB)
-	}
+	dao := blockdao.NewBlockDAO(kvStore, indexers, cfg.Chain.CompressBlock, cfg.DB)
```

Comment:
indexers are appended to blockdao in sync mode. In async mode, we need to add the indexers as subscriber. So you need to revert this change.

Reply:
Get.

Reply:
I handle this at line 163
```
if !cfg.Chain.EnableAsyncIndexWrite {
	indexers = append(indexers, indexer)
}
```

Reply:
Or we need to make systemlog indexer as subsrciber?